spec/lib/scheduler/thread_pool_spec.rb

```diff
@@ -165,6 +174,24 @@
     end
   end
 
+  describe "when thread pool has zero min threads" do
+    it "can quickly process tasks" do
+      # setting idle time to 1000 to ensure that there are maximal delays waiting
+      # for jobs
+      pool = Scheduler::ThreadPool.new(min_threads: 0, max_threads: 5, idle_time: 1000)
+
+      done = Queue.new
+      pool.post { done << :done }
+
+      # should happen in less than 1 second
+      Timeout.timeout(1) { expect(done.pop).to eq(:done) }
+
+      pool.shutdown
+      pool.wait_for_termination
+      expect(pool.stats[:thread_count]).to eq(0)
```

Comment:
I'm not sure if we need these lines cause it is testing `shutdown` logic and don't seem to be related to the test description.

Reply:
happy to amend the description.